### PR TITLE
Add libwebsockets

### PR DESCRIPTION
**Pull request by sodre**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/libwebsockets/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+source:
+  url: https://github.com/warmcat/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 5e731c536a20d9c03ae611631db073f05cd77bf0906a8c30d2a13638d4c8c667
+
+build:
+  number: 0
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x.x') }}
+  skip: True  # [not linux]
```
Comment (sodre) [inline recipes/libwebsockets/meta.yaml]: I tried to compile it on OSX but if failed on OpenSSL issues.

**Comment by sodre** [comment]
@conda-forge/help-c-cpp The libwebsockets recipe is ready for review.

I was not able to get it to compile for macOS.

File: recipes/libwebsockets/build.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+      -DLWS_UNIX_SOCK=ON \
+      -DLWS_WITH_STATIC=OFF \
+      -DLWS_WITHOUT_TESTAPPS=ON \
+      -DLWS_WITH_HTTP_PROXY=ON \
+      -DLWS_WITH_ACCESS_LOG=ON \
+      -DLWS_WITH_LIBUV=ON \
+      -DLWS_WITH_SERVER_STATUS=ON \
+      ..
+
+make -j ${CPU_COUNT}
```
Comment (bgruening) [inline recipes/libwebsockets/build.sh]: Is there a make test available?

Comment (sodre) [reply recipes/libwebsockets/build.sh]: @bgruening, I tried `make test` it did not work.

Comment (scopatz) [reply recipes/libwebsockets/build.sh]: What about a `make check`?

**Comment by scopatz** [comment]
make check would be nice, but I think that this is fine otherwise

File: recipes/libwebsockets/build.sh (hunk 2 above)
Comment (sodre) [reply recipes/libwebsockets/build.sh]: I listed all the targets from cmake for this project. There was nothing along that direction.

**Comment by bgruening** [comment]
Thanks @sodre!

**Pull request merged**